File: src/main/scala/laoflch/debezium/connector/informix/ChangeTable.scala
package laoflch.debezium.connector.informix.integrtest

import io.debezium.relational.TableId


/**
 * A logical representation of change table containing changes for a given source table.
 * There is usually one change table for each source table. When the schema of the source table
 * is changed then two change tables could be present.
 *
 * @author laoflch Luo
 *
 */
object ChangeTable {
  private val CDC_SCHEMA = "syscdcsv1"
}

class ChangeTable(/**
                   * The table from which the changes are captured
                   */
                  val sourceTableId: TableId,

                  /**
                   * The logical name of the change capture process
                   */
                  val captureInstance: String,

                  /**
                   * Numeric identifier of change table in DB2 schema
                   */
                  val changeTableObjectId: Int,

                  /**
                   * A LSN from which the data in the change table are relevant
                   */
                  val startLsn: Lsn,

                  /**
                   * A LSN to which the data in the change table are relevant
                   */
                  var stopLsn: Lsn) // this.changeTableId = sourceTableId != null ? new TableId(sourceTableId.catalog(), CDC_SCHEMA, captureInstance + "_CT") : null;
{
  val changeTableId = if (sourceTableId != null) new TableId(sourceTableId.catalog, ChangeTable.CDC_SCHEMA, captureInstance)
  else null
  /**
   * The table that contains the changes for the source table
   */
  //final private var changeTableId: TableId = null

  def this(captureInstance: String, changeTableObjectId: Int, startLsn: Lsn, stopLsn: Lsn) {
    this(null, captureInstance, changeTableObjectId, startLsn, stopLsn)
  }

  def getCaptureInstance: String = captureInstance

  def getStartLsn: Lsn = startLsn

  def getStopLsn: Lsn = stopLsn

  def setStopLsn(stopLsn: Lsn): Unit = {
    this.stopLsn = stopLsn
  }

  def getSourceTableId: TableId = sourceTableId

  def getChangeTableId: TableId = changeTableId

  def getChangeTableObjectId: Int = changeTableObjectId

  override def toString: String = "Capture instance \"" + captureInstance + "\" [sourceTableId=" + sourceTableId + ", changeTableId=" + changeTableId + ", startLsn=" + startLsn + ", changeTableObjectId=" + changeTableObjectId + ", stopLsn=" + stopLsn + "]"
}

File: src/main/java/io/debezium/connector/informix/ChangeTable.java
/*
 * Copyright Debezium-Informix-Connector Authors.
 *
 * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
 */

package io.debezium.connector.informix;

import io.debezium.relational.TableId;

public class ChangeTable {

    private static final String CDC_SCHEMA = "syscdcsv1";

    /**
     * A LSN from which the data in the change table are relevant
     */
    private final Lsn startLsn;

    /**
     * A LSN to which the data in the change table are relevant
     */
    private Lsn stopLsn;

    /**
     * The table in the CDC schema that captures changes, suitably quoted for Informix
     */
    private final String captureInstance;

    private int changeTableObjectId;
    private TableId sourceTableId;
    private TableId changeTableId;

    public ChangeTable(TableId sourceTableId, String captureInstance, int changeTableObjectId, Lsn startLsn, Lsn stopLsn) {
        this.sourceTableId = sourceTableId;
        this.changeTableObjectId = changeTableObjectId;
        this.startLsn = startLsn;
        this.stopLsn = stopLsn;
        this.captureInstance = captureInstance;
        this.changeTableId = (sourceTableId != null) ? new TableId(sourceTableId.catalog(), ChangeTable.CDC_SCHEMA, captureInstance) : null;
    }

    public ChangeTable(String captureInstance, int changeTableObjectId, Lsn startLsn, Lsn stopLsn) {
        this(null, captureInstance, changeTableObjectId, startLsn, stopLsn);
    }

    public String getCaptureInstance() {
        return captureInstance;
    }

    public Lsn getStartLsn() {
        return startLsn;
    }

    public Lsn getStopLsn() {
        return stopLsn;
    }

    public void setStopLsn(Lsn stopLsn) {
        this.stopLsn = stopLsn;
    }

    public TableId getSourceTableId() {
        return sourceTableId;
    }

    public TableId getChangeTableId() {
        return changeTableId;
    }

    public int getChangeTableObjectId() {
        return changeTableObjectId;
    }

    @Override
    public String toString() {
        return "Capture instance \"" + getCaptureInstance() + "\" [sourceTableId=" + getSourceTableId()
                + ", changeTableId=" + getChangeTableId() + ", startLsn=" + startLsn + ", changeTableObjectId="
                + getChangeTableObjectId() + ", stopLsn=" + stopLsn + "]";
    }
}
